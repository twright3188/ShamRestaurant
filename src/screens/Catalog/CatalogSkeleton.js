import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import { ScrollView, View, StyleSheet, TouchableOpacity } from 'react-native';
import Header from '../../components/Header/index';
import Title from '../../components/Title/index';
import Colors from '../../theme/index';
import { CatalogSkeletonItem } from '../../components/CatalogSkeletonItem'; 

const styles = StyleSheet.create({
  pageStyles: {
    height: '100%',
    width: '100%',
    backgroundColor: Colors.catalogBackground,
  },
  headerStyles: {
    width: '100%',
    backgroundColor: Colors.catalogBackground,
    padding: 15,
    paddingBottom: 9
  },
  headerBlockStyles: {
    paddingLeft: 0,
    paddingRight: 0,
    flexDirection: 'row', 
    justifyContent: 'space-around',
  },
  headerTitleStyles: {
    flex: 1,
  },
});

function CatalogSkeleton() {

  return (
    <ScrollView style={styles.pageStyles}>
      <View style={styles.headerStyles}>
        <Header style={styles.headerBlockStyles}>
          <Title style={styles.headerTitleStyles} color={Colors.coloredButtonBackgroundColor}>Catalog</Title>
          <TouchableOpacity>
            <Icon name="ios-chatbubbles" size={30} color={Colors.activeIcon} />
          </TouchableOpacity>
        </Header>
      </View>
      <CatalogSkeletonItem />
      <CatalogSkeletonItem />
      <CatalogSkeletonItem />
      <CatalogSkeletonItem />
      <CatalogSkeletonItem />
      <CatalogSkeletonItem />
      <CatalogSkeletonItem />
    </ScrollView>
  );
}

export default CatalogSkeleton;