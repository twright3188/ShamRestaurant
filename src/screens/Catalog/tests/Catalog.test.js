import React from 'react';
import renderer from 'react-test-renderer';
import CategoryListContainer from '../index';
import CatalogSkeleton from '../CatalogSkeleton';

describe('catalog tests', () => {
    it('catalog screen renders correctly', () => {
        const tree = renderer.create(<CategoryListContainer />).toJSON();
        expect(tree).toMatchSnapshot(); 
    });
    it('catalog loader renders correctly', () => {
        const tree = renderer.create(<CatalogSkeleton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});