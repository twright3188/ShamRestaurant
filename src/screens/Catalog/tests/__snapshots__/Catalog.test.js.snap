// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalog tests catalog loader renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "padding": 15,
          "paddingBottom": 9,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 24,
              "width": "100%",
            },
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "paddingLeft": 0,
              "paddingRight": 0,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignItems": "center",
                "color": "#222624",
                "flexWrap": "wrap",
                "fontFamily": "Helvetica Neue",
                "fontWeight": "bold",
                "justifyContent": "center",
                "position": "relative",
                "width": "100%",
              },
              Object {
                "fontSize": 35,
                "lineHeight": 39.97,
              },
              Object {
                "flex": 1,
              },
            ]
          }
        >
          Catalog
        </Text>
        <View
          accessible={true}
          clickable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#399AE5",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`catalog tests catalog screen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "padding": 15,
          "paddingBottom": 9,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 24,
              "width": "100%",
            },
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "paddingLeft": 0,
              "paddingRight": 0,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignItems": "center",
                "color": "#222624",
                "flexWrap": "wrap",
                "fontFamily": "Helvetica Neue",
                "fontWeight": "bold",
                "justifyContent": "center",
                "position": "relative",
                "width": "100%",
              },
              Object {
                "fontSize": 35,
                "lineHeight": 39.97,
              },
              Object {
                "flex": 1,
              },
            ]
          }
        >
          Catalog
        </Text>
        <View
          accessible={true}
          clickable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#399AE5",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 72,
          "justifyContent": "center",
          "padding": 24,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F2F7F4",
            "height": 24,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
